core/src/main/scala-3/cats/derived/package.scala

```diff
@@ -0,0 +1,44 @@
+package cats.derived
+
+import cats.*
+
+object semiauto extends
+  CommutativeMonoidDerivation,
+  CommutativeSemigroupDerivation,
+  ContravariantDerivation,
+  EmptyDerivation,
+  EmptyKDerivation,
+  EqDerivation,
+  FoldableDerivation,
+  FunctorDerivation,
+  HashDerivation,
+  InvariantDerivation,
+  MonoidDerivation,
+  MonoidKDerivation,
+  OrderDerivation,
+  PartialOrderDerivation,
+  ReducibleDerivation,
+  SemigroupDerivation,
+  SemigroupKDerivation,
+  ShowDerivation,
+  TraverseDerivation,
+  Instances:
```

Comment:
I think we should remove those traits and move the definitions here (or top level in the `derived` package)

Reply:
Yeah maybe top level so you import `cats.derived.given` for using `deriving` and `cats.derived.auto.given` for magic

Reply:
It would be `import cats.derived.*` though because the extension methods are not given.
Or we could put them in a `given` specifically for that purpose 🤔

Reply:
I think an extension automatically goes into given scope? Or at least I just changed the tests to use `cats.derived.given` and that worked

Reply:
Neat 👍